File: examples/swift/constructs/pipe_operator.swift
// Ported from examples/lumen/constructs/pipe_operator.lm by scripts/port_examples.py; edit the Lumen original, not this file.
func double(x: Int) -> Int {
    return x * 2
}

func add_one(x: Int) -> Int {
    return x + 1
}

func square(x: Int) -> Int {
    return x * x
}

print("Test: Pipe Operator")
print("Without pipe: square(add_one(double(5)))")
print(square(x: add_one(x: double(x: 5))))
print("With pipe: 5 |> double() |> add_one() |> square()")
let result = square(x: add_one(x: double(x: 5)))
print(result)
print("10 |> double():")
print(double(x: 10))
func multiply(a: Int, b: Int) -> Int {
    return a * b
}

print("3 |> double():")
let x = double(x: 3)
print(multiply(a: x, b: 2))

File: examples/rust/constructs/pipe_operator.rs
// Ported from examples/lumen/constructs/pipe_operator.lm by scripts/port_examples.py; edit the Lumen original, not this file.
fn double(x: i64) -> i64 {
    return x * 2;
}

fn add_one(x: i64) -> i64 {
    return x + 1;
}

fn square(x: i64) -> i64 {
    return x * x;
}

fn multiply(a: i64, b: i64) -> i64 {
    return a * b;
}

fn main() {
    println!("Test: Pipe Operator");
    println!("Without pipe: square(add_one(double(5)))");
    println!("{}", square(add_one(double(5))));
    println!("With pipe: 5 |> double() |> add_one() |> square()");
    let result = square(add_one(double(5)));
    println!("{}", result);
    println!("10 |> double():");
    println!("{}", double(10));
    println!("3 |> double():");
    let x = double(3);
    println!("{}", multiply(x, 2));
}
